File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> myMap;
        for (auto n : nums) {
            myMap[n] += 1;
        }
        vector<pair<int, int>> s(myMap.begin(), myMap.end());
        sort(s.begin(), s.end(), [](const auto& a, const auto& b) {
        return a.second > b.second; // '>' for reverse order
        });
        vector<int> res;
        for (int i = 0; i < k; i++) {
            res.push_back(s[i].first);
        }
        return res;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-4.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        res = -10 ** 9
        curTot = 0
        i = 0
        while (i < len(nums)):
            if nums[i] >= 0:
                curTot += nums[i]
            else:
                if i == len(nums) - 1:
                    break
                elif curTot + nums[i] >= nums[i + 1]:
                    curTot += nums[i]
                else:
                    curTot = max(nums[i + 1], curTot + nums[i] + nums[i + 1])
                    i += 1
            i += 1
            res = max(res, curTot)
        return res

        